core/src/main/java/org/apache/servicecomb/core/transport/AbstractTransport.java

```diff
@@ -160,4 +161,83 @@ public Object parseAddress(String address) {
     }
     return new URIEndpointObject(address);
   }
+
+  /**
+   * Handles the request timeout configurations.
+   * 
+   * @param invocation
+   *            invocation of request
+   * @return configuration value
+   */
+  public static long getReqTimeout(Invocation invocation) {
+    long value = 0;
+    String config;
+
+    // get the config base on priority. operationName-->schema-->service-->global
+    String operationName = invocation.getOperationName();
+    String schema = invocation.getSchemaId();
+    String serviceName = invocation.getMicroserviceName();
+
+    config = CONSUMER_REQUEST_TIMEOUT + "." + serviceName + "." + schema + "." + operationName;
```

Comment:
try logic for every invocation is not a good idea
it's not better than liubao's suggestion, but more complex.

Reply:
i just read and passed operation name/schema id/service name to the method now. any suggestion on how to get operation name/schema/service name without invocation object ?